hudi-common/src/main/java/org/apache/hudi/common/util/ClosableIterator.java

```diff
@@ -24,8 +24,29 @@
  * An iterator that give a chance to release resources.
  *
  * @param <R> The return type
+ *
+ * TODO move under common.util.collection
  */
```

Comment:
Not a big fan of `TODO`s, people rarely pay attention and fix it, just my personal opinion, either fix it in this PR or create a JIRA issue to track it.

Reply:
I get your point, but not everyone is the same -- i do follow-up on the TODOs so all in all i find this still helpful even when there's no Jira ticket associated (it's a minor change that doesn't seem to be worth creating a Jira for it)

Reply:
This one actually i've left while doing another change, now since this PR dedicated to the cleanup i will take it up in this one